lib/auth.coffee

```diff
@@ -29,6 +30,23 @@ getAuth = (deps, opts) ->
 	###
 	exports.twoFactor = twoFactor
 
+	userDetails = null
+	exports.getUserDetails = ->
+		return new Promise (resolve, reject) ->
+			if(userDetails)
+				resolve userDetails
+			else
+				request.send
+					method: 'GET'
+					url: '/user/v1/whoami'
+					baseUrl: apiUrl
+				.get('body')
+				.then (body) ->
+					userDetails = body
+					resolve userDetails
+				.catch ->
+					reject undefined
```

Comment:
Resolve and reject should have brackets here too.

Reject should not be called with undefined though! The catch function will be called with an error - you should pass the error to reject like:

```coffeescript
.catch (e) ->
  reject(e)
```

Conveniently, that can be shortened to just `.catch(reject)`.

Alternatively, maybe you meant to run `resolve(undefined)`? I don't think that's right though.

Reply:
I guess that we are going with the `Promise.try` approach, so we might end up w/o a `.catch` if the error needs to be propagated (which from the rest of the code seems to be the case) or with a `.catchReturn` if it needs to be silenced.